weather.go

```diff
@@ -53,18 +51,17 @@ func CliOutput(token, location string) (output string) {
 	}
 	json.Unmarshal(weatherString, &r)
 
-	output = string(fmt.Sprintf("weather: %s\ncelcius: %v\n", r.OneWord, r.Celcius))
+	output = string(fmt.Sprintf("city: %s\nweather: %s\ncelcius: %v\n", r.City, r.OneWord, r.Celcius))
 	return output
 }
 
-func CallUrl(token, location string) http.Response {
+func CallUrl(token string, location *strings.Builder) http.Response {
```

Comment:
It doesn't seem quite right to return `http.Response` here—it feels like a leaky abstraction. What do we end up doing with this value? Just reading its `Body` as an `io.Reader`, right? So how do you feel about returning `io.Reader` here instead?